create/application_test.go

```diff
@@ -238,6 +236,92 @@ func TestApplicationWait(t *testing.T) {
 	}
 }
 
+func TestApplicationBuildFail(t *testing.T) {
+	cmd := applicationCmd{
+		Wait:        true,
+		WaitTimeout: time.Second * 5,
+		Name:        "some-name",
+	}
+	namespace := "default"
+
+	build := &apps.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "any-name",
+			Namespace: namespace,
+			Labels: map[string]string{
+				util.ApplicationNameLabel: cmd.Name,
+			},
+		},
+	}
+
+	client, err := test.SetupClient(build)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var buf bytes.Buffer
+	out, err := output.NewLogOutput(&buf, log.Mode("default"), &output.LogOutputOptions{
+		NoLabels: true, ColoredOutput: false, Timezone: time.Local,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// fill our logs so we have more than errorLogLines
+	logString := "does not compute!"
+	buildLog := []string{}
+	for i := 0; i < errorLogLines+30; i++ {
+		buildLog = append(buildLog, logString)
+	}
+	client.Log = &log.Client{Client: log.NewFake(t, time.Now(), buildLog...), StdOut: out}
+
+	ctx := context.Background()
+
+	// to test the wait we create a ticker that continously updates our
+	// resources in a goroutine to simulate a controller doing the same
+	ticker := time.NewTicker(100 * time.Millisecond)
+	done := make(chan bool)
+	errors := make(chan error, 1)
+
+	go func() {
+		for {
+			select {
+			case <-done:
+				close(errors)
+				return
+			case <-ticker.C:
+				app := &apps.Application{ObjectMeta: metav1.ObjectMeta{
+					Name:      cmd.Name,
+					Namespace: namespace,
+				}}
+
+				if err := setResourceCondition(ctx, client, app, runtimev1.ReconcileSuccess()); err != nil {
+					errors <- err
+				}
+
+				build.Status.AtProvider.BuildStatus = buildStatusError
+				if err := client.Status().Update(ctx, build); err != nil {
+					errors <- err
+				}
+			}
+		}
+	}()
+
+	if err := cmd.Run(ctx, client); err == nil {
+		t.Fatal("expected build error, got nil")
+	}
```

Comment:
maybe that is just me but I do not see these last 20 lines of the log after executing this block. My output:
```bash
Creating a new application
 ✓ created application "some-name" 🏗
 ⠋ waiting for build to start ⏳
 ✗ waiting for build to start ⏳

Your build has failed with status "error". Here are the last 20 lines of the log:
```
(zero log lines)

Reply:
Yeah that is to be expected in the test output as we redirect the logs to a `bytes.Buffer` so we can do some checks on it.

Reply:
true 🤦